bindings/python/iree/runtime/hal.cc

```diff
@@ -61,23 +308,65 @@ void SetupHalBindings(pybind11::module m) {
       .value("HOST_LOCAL", IREE_HAL_MEMORY_TYPE_HOST_LOCAL)
       .value("DEVICE_VISIBLE", IREE_HAL_MEMORY_TYPE_HOST_VISIBLE)
       .value("DEVICE_LOCAL", IREE_HAL_MEMORY_TYPE_DEVICE_LOCAL)
-      .export_values();
+      .export_values()
+      .def("__or__",
+           [](enum iree_hal_memory_type_bits_t self,
+              enum iree_hal_memory_type_bits_t other) { return self | other; })
+      .def("__and__",
+           [](enum iree_hal_memory_type_bits_t self,
+              enum iree_hal_memory_type_bits_t other) { return self & other; });
+
+  py::enum_<enum iree_hal_buffer_compatibility_bits_t>(m, "BufferCompatibility")
+      .value("NONE", IREE_HAL_BUFFER_COMPATIBILITY_NONE)
+      .value("ALLOCATABLE", IREE_HAL_BUFFER_COMPATIBILITY_ALLOCATABLE)
+      .value("IMPORTABLE", IREE_HAL_BUFFER_COMPATIBILITY_IMPORTABLE)
+      .value("EXPORTABLE", IREE_HAL_BUFFER_COMPATIBILITY_EXPORTABLE)
+      .value("QUEUE_TRANSFER", IREE_HAL_BUFFER_COMPATIBILITY_QUEUE_TRANSFER)
+      .value("QUEUE_DISPATCH", IREE_HAL_BUFFER_COMPATIBILITY_QUEUE_DISPATCH)
+      .export_values()
+      .def("__or__",
+           [](enum iree_hal_buffer_compatibility_bits_t self,
+              enum iree_hal_buffer_compatibility_bits_t other) {
+             return self | other;
+           })
+      .def("__and__", [](enum iree_hal_buffer_compatibility_bits_t self,
+                         enum iree_hal_buffer_compatibility_bits_t other) {
+        return self & other;
+      });
+
   py::enum_<enum iree_hal_buffer_usage_bits_t>(m, "BufferUsage")
       .value("NONE", IREE_HAL_BUFFER_USAGE_NONE)
       .value("CONSTANT", IREE_HAL_BUFFER_USAGE_CONSTANT)
       .value("TRANSFER", IREE_HAL_BUFFER_USAGE_TRANSFER)
       .value("MAPPING", IREE_HAL_BUFFER_USAGE_MAPPING)
       .value("DISPATCH", IREE_HAL_BUFFER_USAGE_DISPATCH)
       .value("ALL", IREE_HAL_BUFFER_USAGE_ALL)
-      .export_values();
+      .export_values()
+      .def("__or__",
```

Comment:
neat

Reply:
Could be neater/more transparent but good enough for now.